plugins/woocommerce/includes/class-wc-structured-data.php

```diff
@@ -577,4 +577,31 @@ public function generate_order_data( $order, $sent_to_admin = false, $plain_text
 
 		$this->set_data( apply_filters( 'woocommerce_structured_data_order', $markup, $sent_to_admin, $order ), true );
 	}
+
+	/**
+	 * Check if a GTIN is valid.
+	 * A GTIN is valid if:
+	 * Is a positive numeric string with a length of 8,12,13 or 14.
+	 *
+	 * @see https://schema.org/gtin
+	 * @param string $gtin The GTIN to check.
+	 * @return bool True if valid. False otherwise.
+	 */
+	public function is_valid_gtin( $gtin ) {
+		return is_string( $gtin ) && preg_match( '/^\d{8}$|^\d{12}$|^\d{13}$|^\d{14}$/', $gtin );
+	}
+
+	/**
+	 * Prepare a GTIN input removing everything except numbers.
+	 *
+	 * @param string $gtin The GTIN to prepare.
+	 * @return string Empty string if not GTIN is provided or the string with the replacements.
```

Comment:
Just a small tweak:
```suggestion
	 * @return string Empty string if no GTIN is provided or the string with the replacements.
```

But reading the last part still isn't very clear "the string with the replacements". We are sanitizing / stripping out unwanted characters, we are not replacing anything within the string. Might this sound better?

```suggestion
	 * @return string Empty string if no GTIN is provided or the sanitized GTIN number.
```